src/SwedbankPay.Sdk.Infrastructure/PaymentInstruments/MobilePay/MobilePayPaymentsResource.cs

```diff
@@ -0,0 +1,34 @@
+﻿using SwedbankPay.Sdk.Common;
+using SwedbankPay.Sdk.Extensions;
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace SwedbankPay.Sdk.PaymentInstruments.MobilePay
+{
+    public class MobilePayPaymentsResource : ResourceBase, IMobilePayResource
+    {
+        public MobilePayPaymentsResource(HttpClient httpClient) : base(httpClient)
+        {
+        }
+
+        public async Task<IMobilePayPaymentResponse> Get(Uri id, PaymentExpand paymentExpand = PaymentExpand.None)
+        {
+            if (id == null)
+                throw new ArgumentNullException(nameof(id));
+            Uri url = id.GetUrlWithQueryString(paymentExpand);
+
+            var mobilepaymentResponseDto = await HttpClient.GetAsJsonAsync<MobilePayPaymentResponseDto>(url);
+            return new MobilePayPaymentResponse(mobilepaymentResponseDto, HttpClient);
+        }
+
+        public async Task<IMobilePayPaymentResponse> Create(MobilePayPaymentRequest paymentRequest,
+                                                            PaymentExpand paymentExpand = PaymentExpand.None)

```

Comment:
Codacy found an issue: [Use the overloading mechanism instead of the optional parameters.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-dotnet/pullRequest?prid=6290774)